contribs/gmf/src/directives/mobilemeasurelength.js

```diff
@@ -118,65 +136,85 @@ gmf.MobileMeasureLengthController = function($scope, ngeoDecorateInteraction, $f
    * @type {ol.style.Style|Array.<ol.style.Style>|ol.StyleFunction}
    * @export
    */
-  this.sketchStyle;
-
-  if (this.sketchStyle === undefined) {
-    this.sketchStyle = new ol.style.Style({
-      fill: new ol.style.Fill({
-        color: 'rgba(255, 255, 255, 0.2)'
-      }),
+  this.sketchStyle = new ol.style.Style({
+    fill: new ol.style.Fill({
+      color: 'rgba(255, 255, 255, 0.2)'
+    }),
+    stroke: new ol.style.Stroke({
+      color: 'rgba(0, 0, 0, 0.5)',
+      lineDash: [10, 10],
+      width: 2
+    }),
+    image: new ol.style.RegularShape({
       stroke: new ol.style.Stroke({
-        color: 'rgba(0, 0, 0, 0.5)',
-        lineDash: [10, 10],
+        color: 'rgba(0, 0, 0, 0.7)',
         width: 2
       }),
-      image: new ol.style.RegularShape({
-        stroke: new ol.style.Stroke({
-          color: 'rgba(0, 0, 0, 0.7)',
-          width: 2
-        }),
-        points: 4,
-        radius: 8,
-        radius2: 0,
-        angle: 0
-      })
-    });
-  }
+      points: 4,
+      radius: 8,
+      radius2: 0,
+      angle: 0
+    })
+  });
 
   /**
    * @type {ngeo.interaction.MeasureLengthMobile}
    * @export
    */
-  this.measure = new ngeo.interaction.MeasureLengthMobile($filter('ngeoUnitPrefix'), {
-    precision: this.precision,
-    sketchStyle: this.sketchStyle
-  });
+  this.measure;
 
-  this.measure.setActive(this.active);
-  ngeoDecorateInteraction(this.measure);
 
   /**
    * @type {ngeo.interaction.MobileDraw}
    * @export
    */
+  this.drawInteraction;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.dirty = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.drawing = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.valid = false;
+};
+
+/**
+ * Initialise the controller.
+ */
+gmf.MobileMeasureLengthController.prototype.init = function() {
```

Comment:
Why don't we use the $onInit function from AngularJS ?

Reply:
I think $onInit is exclusive for components, right?

I will refactor it into a component for 2.3

Reply:
Oups, you're right